RUFAS/biophysical/animal/nutrients/nasem_requirements_calculator.py

```diff
@@ -0,0 +1,718 @@
+        )
+        pregnancy_requirement, gravid_uterine_weight_gain = cls._calculate_pregnancy_energy_requirements(
+            lactating, day_of_pregnancy, days_in_milk, gravid_uterine_weight, uterine_weight
+        )
+        lactation_requirement = cls._calculate_lactation_energy_requirements(
+            animal_type, milk_fat, milk_true_protein, milk_lactose, milk_production
+        )
+        dry_matter_intake = cls._calculate_dry_matter_intake(
+            body_weight,
+            mature_body_weight,
+            days_in_milk,
+            lactating,
+            lactation_requirement,
+            parity,
+            body_condition_score_5,
+            ndf_percentage,
+        )
+        protein_requirement = cls._calculate_protein_requirement(
+            lactating,
+            body_weight,
+            frame_weight_gain,
+            gravid_uterine_weight_gain,
+            dry_matter_intake,
+            milk_true_protein,
+            milk_production,
+            ndf_percentage,
+        )
+        calcium_requirement = cls._calculate_calcium_requirement(
+            body_weight,
+            mature_body_weight,
+            day_of_pregnancy,
+            average_daily_gain,
+            dry_matter_intake,
+            milk_true_protein,
+            milk_production,
+            parity,
+        )
+        phosphorus_requirement = cls._calculate_phosphorus_requirement(
+            body_weight,
+            mature_body_weight,
+            animal_type,
+            day_of_pregnancy,
+            average_daily_gain,
+            dry_matter_intake,
+            milk_true_protein,
+            milk_production,
+            parity,
+        )
+        activity_requirement = cls._calculate_activity_energy_requirements(body_weight, housing, distance)
+
+        essential_amino_acids = AMINO_ACID_CALCULATOR.calculate_essential_amino_acid_requirements(
+            animal_type=animal_type,
+            lactating=lactating,
+            body_weight=body_weight,
+            frame_weight_gain=frame_weight_gain,
+            gravid_uterine_weight_gain=gravid_uterine_weight_gain,
+            dry_matter_intake_estimate=dry_matter_intake,
+            milk_true_protein=milk_true_protein,
+            milk_production=milk_production,
+            NDF_conc=ndf_percentage,
+        )
+
+        return NutritionRequirements(
+            maintenance_energy=maintenance_requirement,
+            growth_energy=growth_requirement,
+            pregnancy_energy=pregnancy_requirement,
+            lactation_energy=lactation_requirement,
+            metabolizable_protein=protein_requirement,
+            calcium=calcium_requirement,
+            phosphorus=phosphorus_requirement,
+            dry_matter=dry_matter_intake,
+            activity_energy=activity_requirement,
+            essential_amino_acids=essential_amino_acids,
+        )
+
+    @classmethod
+    def _calculate_maintentance_energy_requirements(
```

Comment:
```suggestion
    def _calculate_maintenance_energy_requirements(
```
Typo, please also fix all references

Reply:
Good catch, thanks Allister